src/rust/inetstack/protocols/icmpv4/datagram/header.rs

```diff
@@ -64,32 +69,19 @@ impl Icmpv4Header {
         let (type_byte, rest_of_header) = self.protocol.serialize();
         buf[0] = type_byte;
         buf[1] = self.code;
-        // Skip the checksum for now.
+        buf[2] = 0;
+        buf[3] = 0;
```

Comment:
Could you rollback these lines?

Reply:
Done.

Reply:
You are still setting these bits. Any specific reason why we cannot rollback?

Reply:
We need to ensure that the bytes 2 and 3 are 0 to calculate the checksum through the header. We filled these bytes using the checksum value in the following lines.